File: src/test/java/com/soapdogg/lispInterpreter/determiner/ExpressionNodeDeterminerTest.java
package com.soapdogg.lispInterpreter.determiner;

import com.soapdogg.lispInterpreter.datamodels.AtomNode;
import com.soapdogg.lispInterpreter.datamodels.ExpressionNode;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

class ExpressionNodeDeterminerTest {

    private ExpressionNodeDeterminer expressionNodeDeterminer;

    @BeforeEach
    void setup() {
        expressionNodeDeterminer = ExpressionNodeDeterminer.newInstance();
    }

    @Test
    void isTrueTest() {
        ExpressionNode expressionNode = Mockito.mock(ExpressionNode.class);

        boolean actual = expressionNodeDeterminer.isExpressionNode(expressionNode);

        Assertions.assertTrue(actual);
    }

    @Test
    void isFalseTest() {
        AtomNode atomNode = Mockito.mock(AtomNode.class);

        boolean actual = expressionNodeDeterminer.isExpressionNode(atomNode);

        Assertions.assertFalse(actual);
    }
}

File: src/test/java/com/soapdogg/lispInterpreter/determiner/ExpressionNodeDeterminerTest.kt
package com.soapdogg.lispInterpreter.determiner

import com.soapdogg.lispInterpreter.datamodels.AtomNode
import com.soapdogg.lispInterpreter.datamodels.ExpressionNode
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito

class ExpressionNodeDeterminerTest {
    private val expressionNodeDeterminer = ExpressionNodeDeterminer()

    @Test
    fun isTrueTest() {
        val expressionNode = Mockito.mock(ExpressionNode::class.java)
        val actual = expressionNodeDeterminer.isExpressionNode(expressionNode)
        Assertions.assertTrue(actual)
    }

    @Test
    fun isFalseTest() {
        val atomNode = Mockito.mock(AtomNode::class.java)
        val actual = expressionNodeDeterminer.isExpressionNode(atomNode)
        Assertions.assertFalse(actual)
    }
}